src/core/quantum_circuit.jl

```diff
@@ -20,8 +20,27 @@ q[2]:
 Base.@kwdef struct QuantumCircuit
     qubit_count::Int
     gates::Vector{AbstractGate} = []
+
+    #Constructor without any gates, for instance: QuantumCircuit(qubit_count=3);
+    function QuantumCircuit(qubit_count::Int,gates::Vector{Any}=[])         
```

Comment:
Will this not match againt something like

`QuantumCircuit(qubit_count=6, gates=Vector{Any}(["not_a_gate"]))`

?

Reply:
the only possible instance of `Vector{Any}` is an empty array. Passing `gate=[ "123 "]` or `gate=[1,2,3]` results in `MethodError`. Should I add a test for those cases?

Reply:
Passing the example, I posted results in a `MethodError` on line 28 when it tries to convert a `String` to an `AbstractGate`. It does match the function with a non-empty Any vector.

```julia
QuantumCircuit(qubit_count=6, gates=Vector{Any}(["not_a_gate"]))
```

Results in
```
ERROR: MethodError: Cannot `convert` an object of type String to an object of type AbstractGate
Closest candidates are:
  convert(::Type{T}, ::T) where T at Base.jl:61
Stacktrace:
  [1] setindex!(A::Vector{AbstractGate}, x::String, i1::Int64)
    @ Base ./array.jl:966
  [2] _unsafe_copyto!(dest::Vector{AbstractGate}, doffs::Int64, src::Vector{Any}, soffs::Int64, n::Int64)
    @ Base ./array.jl:253
  [3] unsafe_copyto!
    @ ./array.jl:307 [inlined]
  [4] _copyto_impl!
    @ ./array.jl:331 [inlined]
  [5] copyto!
    @ ./array.jl:317 [inlined]
  [6] copyto!
    @ ./array.jl:343 [inlined]
  [7] copyto_axcheck!
    @ ./abstractarray.jl:1127 [inlined]
  [8] Array
    @ ./array.jl:626 [inlined]
  [9] convert
    @ ./array.jl:617 [inlined]
 [10] QuantumCircuit(qubit_count::Int64, gates::Vector{Any})
    @ Snowflake ~/codes/anyon/github/Snowflake.jl/src/core/quantum_circuit.jl:28
 [11] #QuantumCircuit#27
    @ ./util.jl:493 [inlined]
 [12] top-level scope
    @ REPL[1]:1
```